enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,891 @@
+  // Currently the authoritative API determines which controller will manage the resource, this will change in a future release.
+  // To ensure the change has been accepted, please verify that the `status.authoritativeAPI` field has been updated to the desired value and that the `Synchronized` condition is present and set to `True`.
+  // +kubebuilder:validation:Enum=MachineAPI;ClusterAPI
+  // +kubebuilder:validation:Default:=MachineAPI
+  // +default:=MachineAPI
+  // +optional
+  AuthoritativeAPI string `json:"authoritativeAPI,omitempty"`
+}
+
+type xxxStatus struct {
+  // authoritativeAPI is the API that is authoritative for this resource.
+  // Valid values are MachineAPI, ClusterAPI and Migrating.
+  // This value is updated by the migration controller to reflect the authoritative API.
+  // Machine API and Cluster API controllers use this value to determine whether or not to reconcile the resource.
+  // When set to Migrating, the migration controller is currently performing the handover of authority from one API to the other.
+  // +kubebuilder:validation:Enum=MachineAPI;ClusterAPI;Migrating
+  // +optional
+  AuthoritativeAPI string `json:"authoritativeAPI,omitempty"`
+
+  // synchronizedGeneration is the generation of the authoritative resource that the non-authoritative resource is synchronised with.
+  // This field is set when the authoritative resource is updated and the sync controller has updated the non-authoritative resource to match.
+  // +kubebuilder:validation:Minimum=0
+  // +optional
+  SynchronizedGeneration int64 `json:"synchronizedGeneration,omitempty"`
+}
+```
+
+These fields will be added to the following resources:
+* Machine
+* MachineSet
+* MachineHealthCheck
+* ControlPlaneMachineSet[^5]
+
+[^5]: The ControlPlaneMachineSet is a Machine API resource that is not present in Cluster API.
+We will extend the `template` field of the ControlPlaneMachineSet to support Cluster API in a separate enhancement.
+These fields will be added when that enhancement is implemented.
+
+#### Admission of non-authoritative resources
+
+A webhook will be introduced that will be used for admission of all Cluster API and Machine API resources
+that are involved in the sync operation.
+
+It will be responsible for validating on updates that any update to a non-authoritative resource is only performed by the sync controller itself.
+
+There will be a small number of fields that are allowed to be updated by the user:
+* `.metadata.labels` - that are not critical to Machine API or Cluster API functionality, and are not in reserved Kubernetes or OpenShift namespaces, and do not conflict with the authoritative resource
+* `.metadata.annotations` - that are not critical to Machine API or Cluster API functionality, and are not in reserved Kubernetes or OpenShift namespaces, and do not conflict with the authoritative resource
```

Comment:
Do we already have a way to enumerate the disallowed labels/annotations? I don't know exactly how many of these there are, but it seems like it might be difficult to be exhaustive.